gap/semigroups/semieunit.gi

```diff
@@ -0,0 +1,715 @@
+              McAlisterTripleSemigroupAction(y)(A, g))) then
+      return true;
+    fi;
+  fi;
+  return false;
+end);
+
+InstallMethod(OneImmutable, "for a McAlister triple semigroup",
+[IsMcAlisterTripleSemigroup],
+function(S)
+  local Y;
+  Y := McAlisterTripleSemigroupSemilattice(S);
+  return MTE(S, DigraphSources(DigraphRemoveLoops(Y))[1], ());
+end);
+
+# (A, g) in S if and only if Ag^-1 is a vertex of the semilattice of S
+InstallMethod(AsList, "for a McAlister triple semigroup",
+[IsMcAlisterTripleSemigroup],
+function(S)
+  local out, g, A, V;
+    out := [];
+    V := DigraphVertexLabels(McAlisterTripleSemigroupSemilattice(S));
+    for g in McAlisterTripleSemigroupGroup(S) do
+      for A in V do
+        if (McAlisterTripleSemigroupAction(S)(A, Inverse(g)) in V) then
+          Add(out, MTE(S, A, g));
+        fi;
+      od;
+    od;
+    SetMcAlisterTripleSemigroupElmList(S, out);
+  return out;
+end);
+
+InstallMethod(String, "for a McAlister triple semigroup",
+[IsMcAlisterTripleSemigroup],
+function(S)
+  local G, X, Y;
+  G := McAlisterTripleSemigroupGroup(S);
+  X := McAlisterTripleSemigroupPartialOrder(S);
+  Y := McAlisterTripleSemigroupSemilattice(S);
+  return Concatenation("McAlisterTripleSemigroup(", String(G), ", ",
+                       String(X), ", ", String(Y), ")");
+end);
+
+#TODO Linebreak hints
+
+InstallMethod(ViewString, "for a McAlister triple semigroup",
+[IsMcAlisterTripleSemigroup],
+function(S)
+  local G;
+  G := McAlisterTripleSemigroupGroup(S);
+  return Concatenation("<McAlister triple semigroup over ", ViewString(G), ">");
+end);
+
+InstallMethod(IsomorphismSemigroups, "for two McAlister triple semigroups",
+[IsMcAlisterTripleSemigroup, IsMcAlisterTripleSemigroup],
+function(S, T)
+  local iso_g, iso_x;
+
+  if not IsIsomorphicDigraph(McAlisterTripleSemigroupSemilattice(S),
+                             McAlisterTripleSemigroupSemilattice(T)) then
+    return fail;
+  fi;
+
+  iso_x := IsomorphismDigraphs(McAlisterTripleSemigroupPartialOrder(S),
+                               McAlisterTripleSemigroupPartialOrder(T));
+
+  if iso_x = fail then
+    return fail;
+  fi;
+
+  iso_g := IsomorphismGroups(McAlisterTripleSemigroupGroup(S),
+                             McAlisterTripleSemigroupGroup(T));
+  if iso_g = fail then
+    return fail;
+  fi;
+
+  if ForAll(McAlisterTripleSemigroupGroup(S),
+         g -> ForAll(DigraphVertices(McAlisterTripleSemigroupPartialOrder(S)),
+         x -> (McAlisterTripleSemigroupAction(S)(x, g)) ^ iso_x
+         = McAlisterTripleSemigroupAction(T)((x ^ iso_x), (g ^ iso_g)))) then
+    return MappingByFunction(S, T, s -> MTE(T, s[1] ^ iso_x, s[2] ^ iso_g));
+  fi;
+
+  return fail;
+end);
+
+InstallMethod(IsIsomorphicSemigroup, "for two McAlister triple semigroups",
+[IsMcAlisterTripleSemigroup and HasGeneratorsOfSemigroup,
+IsMcAlisterTripleSemigroup and HasGeneratorsOfSemigroup],
+function(S, T)
+  if IsomorphismSemigroups(S, T) = fail then
```

Comment:
this function could be shortened to `return IsomorphismSemigroups(S, T) <> fail;`